subprojects/build-init/src/main/groovy/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -63,6 +64,7 @@
     private PathBase distributionBase = PathBase.GRADLE_USER_HOME;
     private String distributionUrl;
     private GradleVersion gradleVersion;
+    private String distType;
```

Comment:
Yes, you cannot use it on `DistributionLocator` but I think it is still valid using it only in Wrapper. 
Replacing 
```
getDistributionFor(version, VALID_TYPES.iterator().next())
```
by
```
getDistributionFor(version, BIN)
```
already should be worth it.